dash/dash.py

```diff
@@ -1550,25 +1559,49 @@ def run_server(
             ]
             raise
 
-        if self._dev_tools.silence_routes_logging:
-            # Since it's silenced, the address doesn't show anymore.
+        # so we only see the "Running on" message once with hot reloading
+        # https://stackoverflow.com/a/57231282/9188800
+        if os.getenv("WERKZEUG_RUN_MAIN") != "true":
             ssl_context = flask_run_options.get("ssl_context")
-            self.logger.info(
-                "Running on %s://%s:%s%s",
-                "https" if ssl_context else "http",
-                host,
-                port,
-                self.config.requests_pathname_prefix,
-            )
+            protocol = "https" if ssl_context else "http"
+            path = self.config.requests_pathname_prefix
+
+            if proxy:
+                served_url, proxied_url = map(urlparse, proxy.split("::"))
+
+                def verify_url_part(served_part, url_part, part_name):
+                    if served_part != url_part:
+                        raise ProxyError(
```

Comment:
Are we happy with proxy errors throwing, or would it be better to just show a warning but keep going?

Reply:
I'm not sure I understand this: If we are to throw an error if the `served_url` portion of `DASH_PROXY` does not match what's already available in protocol/host/port, why force the user to provide it in the first place? Why not just reuse what we already have and use `DASH_PROXY` only for the proxied url part?

Reply:
`DASH_PROXY` presumably comes from some system configuration at a level that's inaccessible to Dash itself, it's just being reported by the environment as something Dash needs to obey.

But I guess that brings up the option to do the reverse of what you're suggesting: ignore `host` and `port` when we have a `DASH_PROXY` - maybe throw a warning or an error if the user tries to override those and they conflict with the proxy, otherwise just run with the values indicated by the proxy.

I don't think we can do that with the protocol, because if you're using https you also need to provide the cert and key. But I get the impression (@gzork can you confirm this?) that at least in our own use of such proxies, encryption is handled by the proxy itself so `DASH_PROXY` will look like `http://<local>::https://<public>`. So if the user _did_ provide `ssl_context` when the proxy indicates `http` it would be an error but that's unlikely to happen accidentally. And the reverse won't happen if the proxy is handling encryption and this is correctly reflected in `DASH_PROXY`.

Reply:
> at least in our own use of such proxies, encryption is handled by the proxy itself so DASH_PROXY will look like http://<local>::https://<public>.

That's correct, in our case the encryption is handled by `haproxy` on the Dash Enterprise server, and I think it's safe to generally also assume that it will be handled by the proxy itself, a note either in the docs or elsewhere would still be good though.